lib/src/layer/polygon_layer/polygon_layer.dart

```diff
@@ -135,17 +135,16 @@ class _PolygonLayerState extends State<PolygonLayer> {
     return List<_ProjectedPolygon>.generate(
       polygons.length,
       (i) {
-        final polygon = polygons[i];
-        final holes = polygon.holePoints;
+        final projectedPolygon = polygons[i];
+        final holes = projectedPolygon.holePoints;
 
-        return _ProjectedPolygon._(
-          polygon: polygon.polygon,
-          points: simplifyPoints(
-            points: polygon.points,
+        return projectedPolygon
+          ..points = simplifyPoints(
```

Comment:
Most importantly, is this even correct? I feel like we're having the same conversation as over the Polygons.
What I understand:
  1. There's a list will all the projected points
  2. Then you compute a simplified list depending on the zoom level based on (1)

Now (2) mutates the list, so I would expect that when you re-simplify for another zoom level you do it off an already simplified list.

Besides correctness, I'm also not sure how much this does in terms of GC pressure.

```dart
   var list = [1,2,3,4,5];
   list = [6,7,8,9,0];
```

is  the same as 

```dart
   final list1 = [1,2,3,4,5];
   final list2 = [6,7,8,9,0];
```

in terms of allocations and GC'ing.

Reply:
Ah, of course, something I missed. Now that we're mutating the original one, that will be used again and again in following processes.

I'm aware that the bottom two are the same, but now we aren't allocating an entirely new `_ProjectedPolygon` every time.

Ok, ignore me, I'll revert that commit, then I think this is all good to go.